### PR TITLE
Fix circuit validation

### PR DESCRIPTION
**Pull request by asanin-epfl**
- Move 'components' errors to BbpError
- Consider '@library' enum when checking files

### CODE REVIEW

**Comment by asanin-epfl** [comment]
Those problems appeared when I was validating our circuits.

File: bluepysnap/circuit_validation.py [hunk 1]
```diff
@@ -233,18 +233,21 @@ def _check_rotations():
     _check_rotations()
     components = config['components']
     errors += _check_components_dir('morphologies_dir', components)
-    errors += _check_components_dir('mechanisms_dir', components)
     errors += _check_components_dir('biophysical_neuron_models_dir', components)
     if errors:
         return errors
+    library = group['@library']
```
Comment (tomdele) [inline bluepysnap/circuit_validation.py:239]: library does not necessary exist in the files.

**Comment by tomdele** [comment]
>Those problems appeared when I was validating our circuits.

From the new circuit building ?

File: bluepysnap/circuit_validation.py (hunk 1 above)
Comment (asanin-epfl) [reply bluepysnap/circuit_validation.py]: Thank you, I missed it.

**Comment by asanin-epfl** [comment]
> From the new circuit building ?

It depends on what do you mean by the new circuit building. I've been validating the reindexed one.

**Comment by tomdele** [comment]
> I've been validating the reindexed one.
ok !

File: bluepysnap/circuit_validation.py [hunk 2]
```diff
@@ -233,18 +233,18 @@ def _check_rotations():
     _check_rotations()
     components = config['components']
     errors += _check_components_dir('morphologies_dir', components)
-    errors += _check_components_dir('mechanisms_dir', components)
     errors += _check_components_dir('biophysical_neuron_models_dir', components)
     if errors:
         return errors
+    holder = group['@library'] if '@library' in group else group
```
Comment (tomdele) [inline bluepysnap/circuit_validation.py:239]: nice !

Comment (asanin-epfl) [reply bluepysnap/circuit_validation.py]: Actually not :) It didn't work out well :(

Comment (asanin-epfl) [reply bluepysnap/circuit_validation.py]: It's good to have tests

Comment (tomdele) [reply bluepysnap/circuit_validation.py]: hahaha tests are better than I am...

File: tests/data/circuit_config.json [hunk 3]
```diff
@@ -6,7 +6,6 @@
   },
   "components": {
     "morphologies_dir": "$COMPONENT_DIR/morphologies",
-    "mechanisms_dir": "$COMPONENT_DIR/mechanisms_dir",
```
Comment (tomdele) [inline tests/data/circuit_config.json:9]: Do you know what is the initial purpose of this ? 
The specs are not super clear and I am NEURON noob

Comment (asanin-epfl) [reply tests/data/circuit_config.json:9]: It's actually a good thing to have. It is a dir for storing `.mod` files. So the sonata circuit is self-contained. All you need is to compile those `.mod` files. It's not a required field. Unfortunately we didn't implement it yet. I hope at some point we will.

Comment (tomdele) [reply tests/data/circuit_config.json:9]: ok thanks for the info !

**Pull request merged**